source/GStreamerMSEMediaPlayerClient.cpp

```diff
@@ -32,20 +32,36 @@ const int64_t segmentStartMaximumDiff = 1000000000;
 
 GStreamerMSEMediaPlayerClient::GStreamerMSEMediaPlayerClient(
     const std::shared_ptr<firebolt::rialto::client::ClientBackendInterface> &ClientBackend)
-    : mClientBackend(ClientBackend), mPosition(0), mDuration(0), mIsConnected(false),
-      mMaxWidth(DEFAULT_MAX_VIDEO_WIDTH), mMaxHeight(DEFAULT_MAX_VIDEO_HEIGHT), mVideoRectangle{0, 0, 1920, 1080}
+    : mClientBackend(ClientBackend), mPosition(0), mDuration(0), mIsConnected(false), 
+      mMaxWidth(DEFAULT_MAX_VIDEO_WIDTH), mMaxHeight(DEFAULT_MAX_VIDEO_HEIGHT), mVideoRectangle{0, 0, 1920, 1080}, mPullDataStopped(false)
 {
     mBackendQueue.start();
 }
 
 GStreamerMSEMediaPlayerClient::~GStreamerMSEMediaPlayerClient()
+{
+    if(!mPullDataStopped)
+    {
+        stopPullingData();
+    }
+}
+
+void GStreamerMSEMediaPlayerClient::stopPullingData()
```

Comment:
This function stopsbufferPuller msg queues (which pull data), BUT it also stops mBackendQueue, which is the main event loop used to any communication with the Rialto's backend.
It means that, _stopPullingData_ name is inaccurate. Rename it to _stopMessageQueues_ or something like that, please.

Reply:
done